### PR TITLE
MultiNode Tradefed: Refactoring and userdata reset

### PR DESCRIPTION
**Pull request by kateyy**
This branch is on top of #15 - please review the two first commits in the other pull request first.

New (optional) feature added here: After a Tradefed session finished and before triggering a `--retry` run, boot all DUTs into fastboot and flash a userdata image to reset them to a clean state. This reduces side effects some (CTS) test module have on others, when they don't undo changes that happen during test executions. Finally, this helps to produce more reliable test results with a lower number of retry runs.

This PR also contains a list of cleanup/refactoring commits. Let me know if you prefer having them in a separate PR.

### CODE REVIEW

**Comment by chase-qi** [comment]
Hi @kateyy, commit 'Add MultiNode TradeFed tests for --shards and with session retry' has been merged already, could you rebase the PR against master? And please fix the shellcheck https://travis-ci.com/Linaro/test-definitions/builds/101075823#L221

**Comment by kateyy** [comment]
> Hi @kateyy, commit 'Add MultiNode TradeFed tests for --shards and with session retry' has been merged already, could you rebase the PR against master? And please fix the shellcheck https://travis-ci.com/Linaro/test-definitions/builds/101075823#L221

Done.

File: automated/lib/android-test-lib [hunk 1]
```diff
@@ -33,8 +33,6 @@ adb_debug_info() {
 }
 
 initialize_adb() {
-    # enable debug
-    set -x
```
Comment (chase-qi) [inline automated/lib/android-test-lib]: +1. I checked existing sigle node tradefed test definitions. We run setup.sh(. ./setup.sh) with the same context from run.sh, and LAVA runs run.sh with sh '-ex' enabled by default, means the trace option should be enabled already.

File: automated/android/multinode/tradefed/tradefed-multinode.yaml [hunk 2]
```diff
@@ -83,7 +78,7 @@ run:
         # available in the CTS execution scripts.
         - |
             exec_result=0
-            sudo -u testuser env "PATH=${PATH}" ./tradefed-multinode.sh \
```
Comment (chase-qi) [inline automated/android/multinode/tradefed/tradefed-multinode.yaml]: The testuser was introduced for vts, vts didn't work with root for some reasons. I am not sure if it still is the case now. As using root for both will make things simpler, I tend to add +1 for this commit. We can revisit it when we  introduce vts to multinode testdef.

File: automated/lib/py_util_lib.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+"""Shared Python 3 utility code.
```
Comment (chase-qi) [inline automated/lib/py_util_lib.py]: We already have `automated/lib/py_test_lib.py` for the same purpose. Is there a reason to add a new one?

**Comment by chase-qi** [comment]
@kateyy ping

File: automated/lib/py_util_lib.py (hunk 3 above)
Comment (chase-qi) [reply automated/lib/py_util_lib.py]: Sounds good to me.

**Pull request merged**